### ISSUE TITLE
Issue downloading video

### ISSUE THREAD

**Issue by Kevin-Prichard** [traumedy/Video-Download#1]
Hi there,

In brief- I tried to download a video where the max format is 1080p60, and it said "Requested format is not available."  I'm on Ubuntu 22.04 (`uname -a` => "Linux ganymede 6.8.0-45-generic #45~22.04.1-Ubuntu SMP PREEMPT_DYNAMIC Wed Sep 11 15:25:05 UTC 2 x86_64 x86_64 x86_64 GNU/Linux") and a somewhat older Python (Python 3.11.0rc1).

- my working copy was pulled so it's at the latest commit [a0e0629](https://github.com/traumedy/Video-Download/commit/a0e0629389919959ac0c189fc821102438c04c83)
- after pulling, I updated pypi dependencies with `pip --install -U requirements.txt` and all three deps were updated; unclear if that caused the problem I'm seeing, but generally you want to freeze `requirements.txt` to the versions you developed against (with '=='), as '>=' tends to give your users potentially random future versions of deps that you may not yet have tested against
- Based on the app's log, I tried the same params in bash and got the same error, `yt-dlp -f best.1 -S res:1080 https://www.youtube.com/watch?v=COQY8Dbkp9A`
- I eliminated `-f best.1` and it downloads: `[info] COQY8Dbkp9A: Downloading 1 format(s): 303+251`, and it's the correct resolution (ffprobe shows `Stream #0:0(eng): Video: vp9 (Profile 0), yuv420p(tv, bt709), 1920x1080, SAR 1:1 DAR 16:9, 59.94 fps, 59.94 tbr, 1k tbn, 1k tbc (default)`)
- keeping `-f best.1` and eliminating `-S res:1080` yields `Requested format is not available.` again

I haven't encountered `best.1` in the wild, and don't see it in the docs, you sure it's legit?  I have been using filter predicates to get the same result: `-f 'bv[height<=1080]'` (syn `-f 'bestvideo[height<=1080]'`)

Here's a screenshot after trying a few times.  This one shows 'Third best quality' but I iterated thru 'Best quality' and 'Second best Quality' first.  The particular video I tried ([COQY8Dbkp9A](https://www.youtube.com/watch?v=COQY8Dbkp9A)) goes only to 1080p60, however the 'Max resolution' dropdown doesn't seem to filter based on what the video provides.  I had also selected a few subtitles (all the English ones), maybe that contributed to the problem, unsure.

![Screenshot from 2024-10-17 16-02-55](https://github.com/user-attachments/assets/e8a2fb5f-d8ae-4d20-a455-bede4be2b31b)

**Comment by traumedy** [comment]
Good call on freezing requirements.txt. The just released PySide6 (Qt) 7.0 seems to have a bug related to table borders that I need to look into.

"best.1" IS a valid format string option and is equivalent to "best" (but for programmatic reasons it is convenient to always add the quality qualifier value). From the documentation:

```
You can select the n'th best format of a type by using best<type>.<n>. For example, best.2 will select the 2nd best combined format. Similarly, bv*.3 will select the 3rd best format that contains a video stream.
```

The download failure you received is perhaps a documentation issue. You specified "Audio+video" for the format type which is actually requesting a file with audio and video combined. but the YouTube video you requested has no combined video files on the server side, only separate audio and video files, so it correctly reported that the format you requested is not available.

If you don't select "Specify format" it works and downloads the 'best quality' combination available trying first to download the best quality video file (which may include audio) and merging it with the best quality audio only file. If no such combination exists it downloads the best quality combined file:

```
By default, yt-dlp tries to download the best available quality if you don't pass any options. This is generally equivalent to using -f bestvideo*+bestaudio/best
```

The UI does no filtering of available options based on the URL provided in the 'Single URL' field as this is incompatible with using the 'URL List' functionality. This functionality could be added when the 'List formats' button is used.

**Comment by traumedy** [comment]
Requirements.txt freeze implemented: c49b8e860f14ef9225b3d5037a4759c27cff5ee9
